### PR TITLE
Playerclass/Add skill trees to prefabricated player classes and create testing for them

### LINKED ISSUE #1276: Filling in TODO's for preset prefab classes
Van and I are going to fill out the TODO sections in the class_prefabs.c file in the playerclass module. These TO DO sections are essentially filling out abilities and skills for classes other than the warrior class. The warrior class is the only class in the function that has code for abilities. This includes being creative with three new combat abilities for each rogue, ranger, bard, wizard, and monk alongside damage numbers that balance them out.

### LINKED ISSUE #1370: Creating tests for new playerclass preset prefab

### PR DESCRIPTION
**Pull request by vanolson**
There were a number of `TODO` if statements in class_prefabs.c, which were meant to be filled with skill trees for a specific prefabricated player class. The only one that had already been created was Warrior. We created these missing prefab skill trees, notably, Bard, Mage, Ranger, Rogue, and Wizard, each of which being a three-tiered linear skill tree, with increasing damage dealt by each increasing skill. We also wrote tests to make sure that these skill trees were being properly created an assigned  in test_class_prefabs.c. In addition, we deleted several obsolete classes that are no longer part of the prefabs. The skill trees team will likely be working on bringing back these deleted classes, but since they aren't fully implemented as classes, we also decided to remove their unfinished skill trees at this time.

### CODE REVIEW

**Comment by wellingtj** [comment]
Hi a few comments for my review. Overall this is solid code but I have a few suggestions for how to improve. The third skill for wizard only has the warrior description. The rogue skills have some of the descriptions for bard. You're also missing spaces on the end of the line for the monk class and a few others. Otherwise the prefabs look good and the tests look perfect too. This is great!

Here is more of an idea for where to go next, but definitely not for this code request. It looks like you have 3 skills with one being significantly better than the next. It may be smart to implement some kind of mana cost mechanism. This is more of just a recommendation that should probably be taken up with skill trees or battles, but it's more of where to go next with this. I think some effects should also be added. For example, the wizard can paralyze you, but the paralyze effect isn't implemented in the skill. This would likely require consulting with the skill team so this also is probably not feasible for this pull request, but again it's an idea worth pursuing if possible.

**Review by wellingtj — APPROVED** [review]
I found the review button. Not sure why that was so hard to find. I left my comments on the feed above

File: src/playerclass/src/class_prefabs.c [hunk 1]
```diff
@@ -436,10 +506,68 @@ int class_prefab_add_skills(class_t* class) {
...
+        /* Currently point to null effects */
+        /* Skills */
+        skill_t* skill_0 = skill_new(skill_id++, ACTIVE, "Blinding Charm", 
+                                     "You blinded your opponent.", 1, 100, 
+                                     NULL);
+        skill_t* skill_1 = skill_new(skill_id++, ACTIVE, "Paralyze Spell", 
+                                     "Your opponent is paralyzed!", 1, 150, 
+                                     NULL);
+        skill_t* skill_2 = skill_new(skill_id++, ACTIVE, "Arcane Explosion", 
+                                     "A powerful spell that causes a massive explosion with high area damage!", 1, 200,
```
Comment (lilyehsani) [inline src/playerclass/src/class_prefabs.c]: Split this line at 80 characters to be consistent with the rest of the file.

**Review by lilyehsani — CHANGES_REQUESTED** [review]
Looks great! Just one minor note. Love the skill choices too!

File: src/playerclass/src/class_prefabs.c (hunk 1 above)
Comment (vanolson) [reply src/playerclass/src/class_prefabs.c]: Made the change!

**Review by KabirDubey — APPROVED** [review]
Looks great to me! The skilltrees team will use this same design if we implement more skilltrees for prefabricated classes.

**Review by lilyehsani — APPROVED** [review]
LGTM! Approved.

**Comment by vanolson** [comment]
Katie and I filled the `TODO`-commented skill trees in `class_prefabs.c`. Katie added Rogue, Wizard, and Monk, and I added Bard and Wizard. We also wrote tests for these classes for them in `test_class_prefabs.c`, where I did Bard, Monk, and Ranger, and Katie did Wizard and Rogue. There weren't any major changes that we made after the review, mainly some style changes and error fixes suggested by Jake and Lily. Jake did suggest a few possible additions, but those are larger-scale and are something that could be a new issue in the future. This issue became related to another one from the skill trees team, #1066, which we coordinated with their team.

**Comment by lilyehsani** [comment]
Issue Score: Excellent

Comments:
Great work on this PR!

**Pull request merged**